pkg/controller/kubelet-config/kubelet_config_controller.go

```diff
@@ -547,6 +547,13 @@ func (ctrl *Controller) addFinalizerToKubeletConfig(kc *mcfgv1.KubeletConfig, mc
 			return err
 		}
 
+		// if the finalizer is already set then skip
```

Comment:
Isn't this not just checking if set  but checking if matches mc.name?

Reply:
Is there ever a way that fubalizer set but doesn't match mc.name? If so should that error? (I have no idea just wondering)

Reply:
I don't know that there are any specific cases, but another controller or user could be adding their own finalizer to objects for whatever reason. This check only carrying about our own finalizer seems correct.

(although I'm not sure why we need to use finalizers in this case? what do they provide over using ownerrefs?)

Reply:
I reviewed the code again (it's been awhile). Looks like we can remove the finalizers. Should we do remove the finalizers in this PR? or a different one?

Reply:
Thought some more on this... I think we should fix this bug with this patch, and support a jira story to remove the finalizers.

Reply:
that's fine with me :)